### PR TITLE
Fix two validation bugs

### PR DESCRIPTION
**Pull request by dbrgn**

### CODE REVIEW

File: validator/tests/test_service.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 @pytest.mark.parametrize(['data', 'errormsg'], [
     (None, 'JSON payload missing'),
-    ('[]', 'Payload does not contain a "data" field'),
+    ('42', 'Payload must be a JSON object'),
```
Comment (rnestler) [inline validator/tests/test_service.py:9]: Why did you remove the `'[]'` test case?

Comment (dbrgn) [reply validator/tests/test_service.py:9]: It's not applicable anymore, since we're now doing an `isinstance` check.

**Review by rnestler — APPROVED** [review]
